File: web/src/components/Logs/Message.js
import React from "react";

const renderAction = message => (
  <tr>
    <th>Action</th>
    <td>{message.action}</td>
    <td style={{overflowWrap: 'break-word', wordBreak: 'break-word'}}>{message.item && message.item.url}</td>
    <td colSpan={3}>{message.content}</td>
  </tr>
)

const renderUpdate = message => (
  <tr>
    <th>Update</th>
    <td>{message.scraper}</td>
    <td style={{overflowWrap: 'break-word', wordBreak: 'break-word'}}>{message.item && message.item.url}</td>
    <td>{message.status}</td>
    <td>{message.message}</td>
  </tr>
)


const Message = ({message}) => message.type === 'action' ? renderAction(message) : renderUpdate(message)

export default Message;

File: types/message.go
package types

type WSUpdatePayload struct {
	Time         int64
	Type         string `json:"type"`
	Scraper      string `json:"scraper"`
	Item         Item   `json:"item"`
	Status       string `json:"status"`
	Message      string `json:"message"`
	AddToCartUrl string `json:"addToCartUrl"`
}

type WSActionPayload struct {
	Time    int64
	Type    string `json:"type"`
	Action  string `json:"action"`
	Item    Item   `json:"item"`
	Content string `json:"content"`
}
